integrations/autotools/creating_packages.rst

```diff
@@ -0,0 +1,174 @@
+standard variables like ``$(CPPFLAGS)`` or ``$(CXX)`` to build it:
+
+.. code-block:: make
+
+    SRC = hello.cpp
+    OBJ = $(SRC:.cpp=.o)
+    OUT = libhello.a
+    INCLUDES = -I.
+
+    .SUFFIXES: .cpp
+
+    default: $(OUT)
+
+    .cpp.o:
+        $(CXX) $(INCLUDES) $(CPPFLAGS) $(CXXFLAGS) -c $< -o $@
+
+    $(OUT): $(OBJ)
+        ar rcs $(OUT) $(OBJ)
+
+
+The *conanfile.py* file uses the ``AutoTools`` build helper. This helper defines
+the necessary environment variables with information from dependencies, as well as other variables
+to match the current Conan settings (like ``-m32`` or ``-m64`` based on the Conan ``arch`` setting)
+
+.. code-block:: python
+
+   from conans import ConanFile
+   from conans import tools
+   from conan.tools.gnu import Autotools
+
+
+   class HelloConan(ConanFile):
+       name = "hello"
+       version = "0.1"
+       settings = "os", "compiler", "build_type", "arch"
+       exports_sources = "src/*"
+       generators = "AutotoolsToolchain"
+
+       def build(self):
+           with tools.chdir("src"):
+               atools = Autotools(self)
+               atools.make()
+
+       def package(self):
+           self.copy("*.h", dst="include", src="src")
+           self.copy("*.lib", dst="lib", keep_path=False)
+           self.copy("*.a", dst="lib", keep_path=False)
+
+       def package_info(self):
+           self.cpp_info.libs = ["hello"]
+
+
+
+With this *conanfile.py* you can create the package:
+
+.. code-block:: bash
+
+    $ conan create . user/testing -s compiler=gcc -s compiler.version=4.9 -s compiler.libcxx=libstdc++
+
+
+Using packages
+------------------
+
+Now let's move to the application folder:
+
+.. code-block:: bash
+
+    $ cd ../helloapp
+
+
+There you can also see a *src* folder with a *Makefile* creating an executable:
+
+.. code-block:: make
+
+    SRC = app.cpp
+    OBJ = $(SRC:.cpp=.o)
+    OUT = app
+    INCLUDES = -I.
+
+    .SUFFIXES: .cpp
+
+    default: $(OUT)
+
+    .cpp.o:
+        $(CXX) $(CPPFLAGS) $(CXXFLAGS) -c $< -o $@
+
+    $(OUT): $(OBJ)
+        $(CXX) -o $(OUT)  $(OBJ)  $(LDFLAGS)  $(LIBS)
+
+
+And also a *conanfile.py* very similar to the previous one. In this case adding a ``requires`` and a ``deploy()`` method:
+
+.. code-block:: python
+   :emphasize-lines: 11, 23, 24
+
+   from conans import ConanFile
+   from conans import tools
+   from conan.tools.gnu import Autotools
+
+
+   class AppConan(ConanFile):
+       name = "app"
+       version = "0.1"
+       settings = "os", "compiler", "build_type", "arch"
+       exports_sources = "src/*"
+       requires = "hello/0.1@user/testing"
+       generators = "AutotoolsDeps", "AutotoolsToolchain"
+
+       def build(self):
+           with tools.chdir("src"):
+               atools = Autotools(self)
+               atools.make()
+
+       def package(self):
+           self.copy("*app", dst="bin", keep_path=False)
+           self.copy("*app.exe", dst="bin", keep_path=False)
+
+       def deploy(self):
```

Comment:
Better not use ``deploy()`` method at the moment, it is still to be discussed. Use ``VirtualRunEnv`` to run things

Reply:
Done